### PR TITLE
Us/warehouses

### PR DESCRIPTION
**Pull request by aaabramov**

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -27,7 +27,9 @@
     "toastr": "^2.1.2",
     "angular-animate": "^1.5.3",
     "angular-bootstrap": "^1.2.5",
-    "components-font-awesome": "^4.6.1"
+    "components-font-awesome": "^4.6.1",
+    "d3": "^3.5.17",
+    "iopctrl": "https://github.com/metormote/iopctrl.git"
```
Comment (ITsvetkoFF) [inline bower.json:32]: this lib has 0 releases. so risky!

File: src/client/app/warehouses/supply.controller.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  angular
+    .module('app.warehouses')
+    .controller('SupplyController', SupplyController);
+
+  SupplyController.$inject = ['$stateParams', 'warehouses', 'logger'];
+  /* @ngInject */
+  function SupplyController($stateParams, warehouses, logger) {
+    var vm = this;
+    console.log($stateParams);
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/supply.controller.js:12]: NO console.logs .

File: src/client/app/warehouses/supply.controller.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    vm.updateAmount = function () {
+      var supply = vm.supply;
+      if (vm.newAmount > supply.maxCapacity) {
+        logger.warning(vm.newAmount + ' is greater than ' + supply.supply + ' ' + supply.maxCapacity);
+        return;
+      }
+      var prevValue = supply.amount; // for rollback
+
+      supply.amount = parseInt(vm.newAmount); // for formats like '00008'
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/supply.controller.js:25]: May it be better to do that in success section?

File: src/client/app/warehouses/supply.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<div class="row" ng-if="vmSupply.supply">
+  <!--<div class="col-md-3">-->
+  <!--<h2><i class="fa fa-dropbox"></i>Supply:</h2>-->
+  <!--<h1 class="text-primary">{{vmSupply.supply.supply}}</h1>-->
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/supply.html:4]: necrocode? :)

File: src/client/app/warehouses/warehouses.directives.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+/**
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/warehouses.directives.js:1]: src/client/app/warehouses/warehouses.directives.js
  line 13  col 11  Missing semicolon.             (W033)
  line 19  col 8   Missing semicolon.             (W033)
  line 35  col 15  Strings must use singlequote.  (W109)
  line 36  col 20  Strings must use singlequote.  (W109)
  line 37  col 21  Strings must use singlequote.  (W109)
  line 61  col 19  Strings must use singlequote.  (W109)
  line 62  col 20  Strings must use singlequote.  (W109)
  line 62  col 41  Strings must use singlequote.  (W109)
  line 63  col 24  Strings must use singlequote.  (W109)
  line 63  col 38  Strings must use singlequote.  (W109)
  line 63  col 92  Strings must use singlequote.  (W109)
  line 66  col 19  Strings must use singlequote.  (W109)
  line 67  col 20  Strings must use singlequote.  (W109)
  line 67  col 39  Strings must use singlequote.  (W109)
  line 40  col 17  'iopctrl' is not defined.      (W117)
  line 43  col 18  'iopctrl' is not defined.      (W117)
  line 55  col 22  'iopctrl' is not defined.      (W117)

  ⚠  17 warnings

File: src/client/app/warehouses/supply.html [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      <i class="fa fa-exclamation-triangle" ng-show="vmSupply.isWarning()&& !vmSupply.isOverflow()"></i>
+      <i class="fa fa-stack-overflow" ng-show="vmSupply.isOverflow()"></i>
+    </h3>
+  </div>
+
+  <div class="col-md-6">
+    <form name="update_form" ng-submit="update_form.$valid && vmSupply.updateAmount()">
+      <fieldset class="form-group">
+        <h1>New amount:</h1>
+        <input type="text" ng-model="vmSupply.newAmount" pattern="\d\d+" class="text-input form-control">
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/supply.html:21]: why "\d\d+" ?

File: src/client/app/warehouses/supply.html [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        <input class="btn btn-lg btn-primary" type="submit" value="Update" ng-disabled="!vmSupply.enabledToUpdate()"/>
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/supply.html:22]: `ng-disabled="vmSupply.disabledToUpdate()"` ? :)

File: src/client/app/warehouses/warehouses.controller.js [hunk 8]
```diff
@@ -1,22 +1,30 @@
...
     vm.title = 'Warehouses';
+    vm.supplies = [];
+    vm.location = $location;
+
 
     activate();
 
     function activate() {
-      warehouses.getWarehouses().then(function(data) {
+      warehouses.getWarehouses().success(function (data) {
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/warehouses.controller.js:21]: oh, success is deprecated. use `then`

`The $http legacy promise methods success and error have been deprecated. Use the standard then method instead. If $httpProvider.useLegacyPromiseExtensions is set to false then these methods will throw $http/legacy error.`

File: src/client/app/warehouses/warehouses.html [hunk 9]
```diff
@@ -1,23 +1,24 @@
...
-              <td>{{supply.supply}}</td>
-              <td>{{supply.amount}}</td>
-            </tr>
-            </tbody>
-          </table>
+          <div class="row">
+            <div class="col-md-3">
+              <ul class="nav nav-pills nav-stacked">
+                <li ng-repeat="supply in vmWarehouses.supplies | orderBy:'supply'"
+                    ng-class="{'active': vmWarehouses.location.absUrl().includes(supply.supply.toLowerCase())}">
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/warehouses.html:11]: воу воу воу. http://angular-ui.github.io/ui-router/site/#/api/ui.router.state.directive:ui-sref-active

File: src/client/app/warehouses/warehouses.route.js [hunk 10]
```diff
@@ -26,6 +26,19 @@
...
+      {
+        state: 'warehouses.supply',
+        config: {
+          url: '/:name',
+          templateUrl: 'app/warehouses/supply.html',
+          controller: 'SupplyController',
+          controllerAs: 'vmSupply',
+          title: 'supply',
+          params: {
+            supply: undefined
```
Comment (ITsvetkoFF) [inline src/client/app/warehouses/warehouses.route.js:39]: `supply: undefined` - how it works?

File: src/client/app/warehouses/warehouses.html (hunk 9 above)
Comment (aaabramov) [inline src/client/app/warehouses/warehouses.html:11]: I've already read too much variants.
None of them worked!

Comment (ITsvetkoFF) [inline src/client/app/warehouses/warehouses.html:11]: we may try `ui-sref-active` together. it will be cleaner!

**Pull request closed without merging**